src/Manager.php

```diff
@@ -281,19 +338,15 @@ public function getDefaultRoleNames(): array
 
     public function getDefaultRoles(): array
     {
-        $roles = $this->itemsStorage->getRolesByNames($this->defaultRoleNames);
-        $missingRoles = array_diff($this->defaultRoleNames, array_keys($roles));
-        if (!empty($missingRoles)) {
-            $missingRolesStr = '"' . implode('", "', $missingRoles) . '"';
-
-            throw new DefaultRolesNotFoundException("The following default roles were not found: $missingRolesStr.");
-        }
-
-        return $roles;
+        return $this->filterStoredRoles($this->defaultRoleNames);
     }
 
     public function setGuestRoleName(?string $name): self
     {
+        if ($name !== null && !$this->itemsStorage->roleExists($name)) {
```

Comment:
It add additional request to storage on each manage configuration. I think better remove this check... If user set invalid guest role name, than manager also throw exception, but in another place.

Reply:
https://github.com/yiisoft/rbac/pull/203/commits/4e9bd6fd6519e4bf455c786eb4e27b4a5673267d